.github/workflows/rust.yml

```
@@ -11,15 +11,24 @@ env:
 
 jobs:
   check:
-    runs-on: windows-2019
-    
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [macOS-latest, windows-2019, ubuntu-latest]
     steps:
     - name: install libx11-dev
       run: |
         sudo apt update
         sudo apt install libx11-dev libpango1.0-dev libxkbcommon-dev libxkbcommon-x11-dev
       if: runner.os == 'Linux'
     - uses: actions/checkout@v3
-    - uses: dtolnay/rust-toolchain@stable
+    - uses: dtolnay/rust-toolchain@master
+      with:
+        toolchain: stable
+        components: rustfmt
```

Just a small optimization (to avoid wasting cloud compute resources), I think it's enough to format on just one system (likely linux).

---

Can with a conditional here... I didn't want to do it as a separate job as that means a separate spin up, clone, install of rust, etc and that's not a worthwhile use of resources.

---

Yeah that's exactly what I've thought of as well.

---

> I didn't want to do it as a separate job as that means a separate spin up, clone, install of rust, etc and that's not a worthwhile use of resources.

Btw. isn't that exactly what happens with `matrix` (just that it's some syntactic sugar that abstracts it away as the workflow is similar enough)?

---

One thing I think would be cool is something like (I'm not sure if this is working, it's untested):

```suggestion
        components: ${{ runner.os == 'Linux' && 'rustfmt' || '' }}
```

---

@Philipp-M That did work. Thanks.